### ISSUE TITLE
Complex types are documented as 'object' or 'any'

### ISSUE THREAD

**Issue by benstevens48** [TypeStrong/typedoc#958]
- [x] I have checked [issues with bug label](https://github.com/TypeStrong/typedoc/labels/bug) and found no duplicates

## Expected Behavior
Function defined as
```
expandNodes(items: {nodeOrLink: {id: string}, type: string, expandedType: string | null}[]){
...
}
```
should be documented as
```
expandNodes(items: {nodeOrLink: {id: string}, type: string, expandedType: string | null}[]): void
```

## Actual Behavior
Function defined as
```
expandNodes(items: {nodeOrLink: {id: string}, type: string, expandedType: string | null}[]){
...
}
```
is documented as
```
expandNodes(items: object[]): void
```

## Steps to reproduce the bug
Run typedoc with a ts file containing a class containing the above function. (TypeScript strict mode is on, in case that makes a difference.)

## Environment
  - Typedoc version: 0.14.2
  - Node.js version: 10.15
  - OS: Windows 10
  - TypeScript version 3.2.2

If this is due to a typedoc setting or something then I apologize but I couldn't find such a setting documented anywhere.

**Comment by 9oelM** [comment]
Perhaps related: #953, #901

**Comment by benstevens48** [comment]
Yes, it does look like it might be the same bug, although `excludeNotExported` doesn't seem to make any difference to me.

**Comment by ppoulard** [comment]
* Similar problem with explicit type definition :
```
type CastContext<C> = {
    cast: typeof UNICAST
    client ?: C
} | {
    cast: typeof MULTICAST
    clients: C[]
} | {
    cast: typeof BROADCAST
}
```
is rendered as 
```
CastContext<C>: object | object | object
```
instead of the object description

* Another example :
```
export const UNICAST: unique symbol = Symbol('UNICAST')
```
is correctly rendered as :
```
UNICAST: keyof symbol = Symbol('UNICAST')
```
however, if I refer to it with `typeof` :
```
export type Capabilities = typeof UNICAST | typeof MULTICAST | typeof BROADCAST;
```
it is rendered as :
```
Capabilities: unique symbol | unique symbol | unique symbol
```
which is somewhat useless

**Comment by bobheadxi** [comment]
Another similar problem I ran into in https://github.com/ubclaunchpad/new/issues/22:

```ts
export const clubConfig: {
  memberCount: number;
  currentTeams: Team[];
  featuredTeams: Team[];
} = {
  /**
   * The current or most recent active Launch Pad teams and their primary project.
   * 
   * Each element must satisfy [[`Team`]].
   */
  currentTeams: [ ... ]
```

generates the documentation *without* taking into account `currentTeams: Team[]`:

<img width="944" alt="image" src="https://user-images.githubusercontent.com/23356519/79698047-1b3ec000-823b-11ea-83c3-787448adb6e1.png">

**Comment by Gerrit0** [comment]
Huh, that's interesting. Looks to me like we are completely ignoring the type annotation, which is definitely an issue... and I thought we did look at that...

**Comment by Gerrit0** [comment]
These all appear to be fixed in 0.20.

![image](https://user-images.githubusercontent.com/19329837/100528846-6ee49500-319e-11eb-92c7-5707791c97f2.png)

![image](https://user-images.githubusercontent.com/19329837/100528851-786dfd00-319e-11eb-8351-7ec0aef1d667.png)

![image](https://user-images.githubusercontent.com/19329837/100528852-7e63de00-319e-11eb-9935-8a02618c1541.png)
